server/app/watchlist.py

```diff
@@ -0,0 +1,88 @@
+import os.path
+from sys import stderr
+from configparser import ConfigParser
+
+url_formats = {
+	'youtube': {
+		'template': 'https://{}/feeds/videos.xml?channel_id={}',
+		'default_site': 'www.youtube.com',
+	},
+	'invidious': {
+		'template': 'https://{}/feed/channel/{}',
+		'default_site': 'vid.puffyan.us',
+	},
+}
+
+class Watchlist:
+	def __init__(self, file_path):
+		self.__is_invalid = False
+		parser = ConfigParser()
+		parser.read(file_path)
+		channels = []
+		for section in parser.sections():
+			entry = parser[section]
+
+			if 'id' not in entry:
+				self.__is_invalid = True
+				continue
+			id = entry['id']
+
+			if 'source' not in entry:
+				source = 'youtube'
+			else:
+				source = entry['source'].lower()
+
+			if source not in url_formats:
+				self.__is_invalid = True
+				continue
+
+			if 'site' not in entry:
```

Comment:
what's the difference between source and site and why do we need to store both?

Reply:
It's mainly there for invidious. Since anyone can host their own instance.